### PR TITLE
local mode: support output_path

### PR DESCRIPTION
**Pull request by iquintero**
Can be either file:// or s3:// - This also changes the default
behavior of local mode to use the SDK provided default S3 bucket
if nothing is passed. This makes it easier for customers to create
models in SageMaker too since their Model Artifacts will already be
a tarfile in S3.

*Issue #, if available:*

*Description of changes:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have updated the [changelog](https://github.com/aws/sagemaker-python-sdk/blob/master/CHANGELOG.rst) with a description of my changes (if appropriate)
- [x] I have updated any necessary [documentation](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/estimator.py [hunk 1]
```diff
@@ -104,6 +104,9 @@ def __init__(self, role, train_instance_count, train_instance_type,
 
         self.base_job_name = base_job_name
         self._current_job_name = None
+        if (not self.sagemaker_session.local_mode
+                and output_path and output_path.startswith('file://')):
+            raise RuntimeError('file:// output paths are only supported in Local Mode')
```
Comment (nadiaya) [inline src/sagemaker/estimator.py]: 👌

File: src/sagemaker/local/image.py [hunk 2]
```diff
@@ -201,26 +198,49 @@ def retrieve_artifacts(self, compose_data):
...
+                    sagemaker.local.utils.recursive_copy(host_dir, model_artifacts)
                 elif container_dir == '/opt/ml/output':
-                    sagemaker.local.utils.recursive_copy(host_dir, s3_output_artifacts)
+                    sagemaker.local.utils.recursive_copy(host_dir, output_artifacts)
+
+        # Tar Artifacts -> model.tar.gz and output.tar.gz
+        model_files = [os.path.join(model_artifacts, name) for name in os.listdir(model_artifacts)]
+        output_files = [os.path.join(output_artifacts, name) for name in os.listdir(output_artifacts)]
+        sagemaker.utils.create_tar_file(model_files, os.path.join(compressed_artifacts, 'model.tar.gz'))
+        sagemaker.utils.create_tar_file(output_files, os.path.join(compressed_artifacts, 'output.tar.gz'))
```
Comment (nadiaya) [inline src/sagemaker/local/image.py]: Should that be a function? (all 5 lines)
that seems a bit repetitive.

File: src/sagemaker/local/image.py [hunk 3]
```diff
@@ -201,26 +198,49 @@ def retrieve_artifacts(self, compose_data):
...
+                compressed_artifacts,
+                output_data_config['S3OutputPath'],
+                job_name,
+                self.sagemaker_session)
+
+        _delete_tree(model_artifacts)
+        _delete_tree(output_artifacts)
 
-        return s3_model_artifacts
+        return os.path.join(output_data, 'model.tar.gz')
```
Comment (nadiaya) [inline src/sagemaker/local/image.py]: Since we moved both files should we return both (model and output) too??

Comment (iquintero) [reply src/sagemaker/local/image.py]: Not really,  SageMaker just returns the S3ModelArtifacts - if you want to look at output.tar.gz you basically have to do a replace in the string. This gets sent directly to the local client describeTrainingJob()

**Pull request merged**